packages/combo-box/src/vaadin-combo-box-data-provider-mixin.js

```diff
@@ -110,7 +114,6 @@ export const ComboBoxDataProviderMixin = (superClass) =>
         this.__previousDataProviderFilter = filter;
 
         this.__keepOverlayOpened = true;
-        this._pendingRequests = {};
         this.size = undefined;
         this.clearCache();
```

Comment:
**note:** The `requestContentUpdate` method is called twice during these two lines. This could be optimized by updating the size in the controller instead of the component, leaving `requestContentUpdate` to happen only in `clearCache`.

Reply:
Looks like the tests pass now even with `this.size = undefined;` removed

Reply:
Thanks for the catch. This line is important because it ensures combo-box clears the list when you start filtering. Created https://github.com/vaadin/web-components/pull/7465 to cover this line.